File: A2OJ/Ladder4_Codeforces_Div2A/005_HelpfulMaths/helpfulMaths.java
/*
 * Autor: Frederico Santos Gonçalves
 * Data: 15/01/2026
 *
 * Plataforma: A2OJ
 * Problema: Helpful Maths
 * Link: https://codeforces.com/problemset/problem/339/A
 * Dificuldade: 1
 */

import java.util.ArrayList;
import java.util.Scanner;
import java.util.Collections;

public class helpfulMaths {
  private static final Scanner scanner = new Scanner(System.in);

  public static void main(String[] args) {
    String s = scanner.nextLine();
    char[] S = s.toCharArray();
    
    ArrayList<Integer> nums = new ArrayList<>();
    for(int i = 0; i < s.length(); i++) if(i%2 == 0) nums.add(S[i]-'0');
    Collections.sort(nums);

    for(int i = 0; i < nums.size(); i++) {
      if(i != 0) System.out.print("+");
      System.out.print(nums.get(i));
    }
  }
}

File: A2OJ/Ladder4_Codeforces_Div2A/005_HelpfulMaths/helpfulMaths.c
/*
 * Autor: Frederico Santos Gonçalves
 * Data: 15/01/2026
 *
 * Plataforma: A2OJ
 * Problema: Helpful Maths
 * Link: https://codeforces.com/problemset/problem/339/A
 * Dificuldade: 1
 */

#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#define SUCESSO (0)
#define TAM_MAX (1000)

void bubbleSort(int* array, int tam) {
  for(int i = 0; i < tam; i++) {
    for(int j = 0; j < tam - 1; j++) {
      if(array[j] <= array[j+1]) continue;
      else {
        int temp = array[j];
        array[j] = array[j+1];
        array[j+1] = temp;
      }
    }
  }
}

int main() {
  char s[TAM_MAX];
  scanf("%s%*c", &s);

  int nums[TAM_MAX];
  for(int i = 0; i < strlen(s); i++) {
    if(s[i] != '+') {
      char c[2];
      c[0] = s[i];
      c[1] = '\0';
      nums[i/2] = atoi(c);
    }
  }
  bubbleSort(nums, (strlen(s)/2)+1);

  for(int i = 0; i < (strlen(s)/2)+1; i++) {
    if(i != 0) printf("+");
    printf("%d", nums[i]);
  }

  return SUCESSO;
}
